[crashpad] Linker errors with undefined references after building with vcpkg on Ubuntu

[proc] 执行命令: /usr/local/share/vcpkg/downloads/tools/cmake-3.29.2-linux/cmake-3.29.2-linux-x86_64/bin/cmake --build /home/fxy/demo/Cpp-Examples/build --config Debug --target all --
[build] [1/2  50% :: 1.311] Building CXX object Crashpad/CMakeFiles/crashpad_test.dir/main.cc.o
[build] [2/2 100% :: 1.522] Linking CXX executable /home/fxy/demo/Cpp-Examples/bin-64/Debug/crashpad_test
[build] FAILED: /home/fxy/demo/Cpp-Examples/bin-64/Debug/crashpad_test 
[build] : && /usr/bin/g++ -g  Crashpad/CMakeFiles/crashpad_test.dir/main.cc.o -o /home/fxy/demo/Cpp-Examples/bin-64/Debug/crashpad_test -L/home/fxy/demo/Cpp-Examples/bin-64/libs -Wl,-rpath,/home/fxy/demo/Cpp-Examples/bin-64/libs  vcpkg_installed/x64-linux/debug/lib/libclient.a  vcpkg_installed/x64-linux/debug/lib/libutil.a  vcpkg_installed/x64-linux/debug/lib/libbase.a  vcpkg_installed/x64-linux/debug/lib/libcommon.a  vcpkg_installed/x64-linux/debug/lib/libz.a && cd /home/fxy/demo/Cpp-Examples/build/Crashpad && /usr/local/share/vcpkg/downloads/tools/cmake-3.29.2-linux/cmake-3.29.2-linux-x86_64/bin/cmake -E copy_directory /home/fxy/demo/Cpp-Examples/build/vcpkg_installed/x64-linux/tools/crashpad/ /home/fxy/demo/Cpp-Examples/bin-64/Debug
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libclient.a(client.crashpad_client_linux.o): in function `crashpad::CrashpadClient::InitializeSignalStackForThread()':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crashpad_client_linux.cc:544: undefined reference to `pthread_getspecific'
[build] /usr/bin/ld: /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crashpad_client_linux.cc:567: undefined reference to `pthread_setspecific'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libclient.a(client.crashpad_client_linux.o): in function `crashpad::CrashpadClient::InitializeSignalStackForThread()::$_1::operator()() const':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crashpad_client_linux.cc:536: undefined reference to `pthread_key_create'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.settings.o): in function `crashpad::Settings::InitializeSettings(int)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/settings.cc:406: undefined reference to `crashpad::UUID::InitializeWithNew()'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database_generic.o): in function `crashpad::CrashReportDatabaseGeneric::FinishedWritingCrashReport(std::unique_ptr<crashpad::CrashReportDatabase::NewReport, std::default_delete<crashpad::CrashReportDatabase::NewReport> >, crashpad::UUID*)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:359: undefined reference to `crashpad::FileWriter::Close()'
[build] /usr/bin/ld: /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:368: undefined reference to `crashpad::FileWriter::Close()'
[build] /usr/bin/ld: /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:376: undefined reference to `crashpad::Metrics::CrashReportPending(crashpad::Metrics::PendingReportReason)'
[build] /usr/bin/ld: /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:377: undefined reference to `crashpad::Metrics::CrashReportSize(long)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database_generic.o): in function `crashpad::CrashReportDatabaseGeneric::ReportPath(crashpad::UUID const&, crashpad::CrashReportDatabaseGeneric::ReportState)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:611: undefined reference to `crashpad::UUID::ToString[abi:cxx11]() const'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database_generic.o): in function `crashpad::CrashReportDatabaseGeneric::SkipReportUpload(crashpad::UUID const&, crashpad::Metrics::CrashSkippedReason)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:432: undefined reference to `crashpad::Metrics::CrashUploadSkipped(crashpad::Metrics::CrashSkippedReason)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database_generic.o): in function `crashpad::CrashReportDatabaseGeneric::RequestUpload(crashpad::UUID const&)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:521: undefined reference to `crashpad::Metrics::CrashReportPending(crashpad::Metrics::PendingReportReason)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database_generic.o): in function `crashpad::CrashReportDatabaseGeneric::CleanOrphanedAttachments()':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:792: undefined reference to `crashpad::UUID::InitializeFromString(base::BasicStringPiece<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const&)'
[build] /usr/bin/ld: /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:800: undefined reference to `crashpad::UUID::ToString[abi:cxx11]() const'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database_generic.o): in function `crashpad::CrashReportDatabaseGeneric::RecordUploadAttempt(crashpad::CrashReportDatabase::UploadReport*, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:562: undefined reference to `crashpad::Metrics::CrashUploadAttempted(bool)'
[build] /usr/bin/ld: /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:583: undefined reference to `crashpad::FileReader::Close()'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database_generic.o): in function `crashpad::(anonymous namespace)::UUIDFromReportPath(base::FilePath const&)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:45: undefined reference to `crashpad::UUID::InitializeFromString(base::BasicStringPiece<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const&)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database_generic.o): in function `crashpad::CrashReportDatabaseGeneric::ReadMetadata(base::FilePath const&, crashpad::CrashReportDatabase::Report*)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database_generic.cc:831: undefined reference to `crashpad::UUID::InitializeFromString(base::BasicStringPiece<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const&)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database_generic.o): in function `base::ScopedGeneric<base::FilePath, crashpad::ScopedRemoveFileTraits>::FreeIfNecessary()':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/third_party/mini_chromium/mini_chromium/base/scoped_generic.h:87: undefined reference to `crashpad::ScopedRemoveFileTraits::Free(base::FilePath const&)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database.o): in function `crashpad::CrashReportDatabase::NewReport::Initialize(crashpad::CrashReportDatabase*, base::FilePath const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database.cc:67: undefined reference to `crashpad::UUID::InitializeWithNew()'
[build] /usr/bin/ld: /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database.cc:74: undefined reference to `crashpad::UUID::ToString[abi:cxx11]() const'
[build] /usr/bin/ld: /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database.cc:78: undefined reference to `crashpad::FileWriter::Open(base::FilePath const&, crashpad::FileWriteMode, crashpad::FilePermissions)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database.o): in function `crashpad::CrashReportDatabase::NewReport::Reader()':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database.cc:88: undefined reference to `crashpad::FileReader::Open(base::FilePath const&)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database.o): in function `crashpad::CrashReportDatabase::NewReport::AddAttachment(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database.cc:113: undefined reference to `crashpad::FileWriter::Open(base::FilePath const&, crashpad::FileWriteMode, crashpad::FilePermissions)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database.o): in function `crashpad::CrashReportDatabase::AttachmentsPath(crashpad::UUID const&)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database.cc:183: undefined reference to `crashpad::UUID::ToString[abi:cxx11]() const'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database.o): in function `crashpad::CrashReportDatabase::UploadReport::InitializeAttachments()':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database.cc:136: undefined reference to `crashpad::FileReader::Open(base::FilePath const&)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database.o): in function `crashpad::CrashReportDatabase::UploadReport::Initialize(base::FilePath const&, crashpad::CrashReportDatabase*)':
[build] /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg/../src/4c3e2d1c10-27b97453fa.clean/client/crash_report_database.cc:167: undefined reference to `crashpad::FileReader::Open(base::FilePath const&)'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database.o): in function `std::_MakeUniq<crashpad::FileWriter>::__single_object std::make_unique<crashpad::FileWriter>()':
[build] /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/unique_ptr.h:857: undefined reference to `crashpad::FileWriter::FileWriter()'
[build] /usr/bin/ld: vcpkg_installed/x64-linux/debug/lib/libcommon.a(common.crash_report_database.o): in function `std::_MakeUniq<crashpad::FileReader>::__single_object std::make_unique<crashpad::FileReader>()':
[build] /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/unique_ptr.h:857: undefined reference to `crashpad::FileReader::FileReader()'
[build] collect2: error: ld returned 1 exit status
[build] ninja: build stopped: subcommand failed.
[proc] 命令“/usr/local/share/vcpkg/downloads/tools/cmake-3.29.2-linux/cmake-3.29.2-linux-x86_64/bin/cmake --build /home/fxy/demo/Cpp-Examples/build --config Debug --target all --”已退出，代码为 1
[driver] 生成完毕: 00:00:01.569
[build] 生成已完成，退出代码为 1


<details><summary>CMakeLists.txt</summary>

```
add_executable(crashpad_test main.cc)
target_link_libraries(crashpad_test PRIVATE crashpad::crashpad)

if(CMAKE_HOST_LINUX)
  target_link_libraries(crashpad_test PRIVATE Threads::Threads)
endif()

```
</details>

--- a/ports/crashpad/crashpadConfig.cmake.in
+++ b/ports/crashpad/crashpadConfig.cmake.in
@@ -9,7 +9,7 @@ endif()
 add_library(crashpad INTERFACE)
 add_library(crashpad::crashpad ALIAS crashpad)
 
-set(CRASHPAD_LIBRARIES client util base common)
+set(CRASHPAD_LIBRARIES common client util base)
 
 if(WIN32)
 	target_compile_definitions(crashpad INTERFACE NOMINMAX)

--- a/ports/crashpad/vcpkg.json
+++ b/ports/crashpad/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "crashpad",
   "version-date": "2022-09-05",
-  "port-version": 5,
+  "port-version": 6,
   "description": [
     "Crashpad is a crash-reporting system.",
     "Crashpad is a library for capturing, storing and transmitting postmortem crash reports from a client to an upstream collection server. Crashpad aims to make it possible for clients to capture process state at the time of crash with the best possible fidelity and coverage, with the minimum of fuss."

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1986,7 +1986,7 @@
     },
     "crashpad": {
       "baseline": "2022-09-05",
-      "port-version": 5
+      "port-version": 6
     },
     "crashrpt": {
       "baseline": "1.4.3",

--- a/versions/c-/crashpad.json
+++ b/versions/c-/crashpad.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d3d6e9473bb3047d0ba5db163765218b013809c1",
+      "version-date": "2022-09-05",
+      "port-version": 6
+    },
+    {
       "git-tree": "f7161b19eece4ad2643d6b2baafdb5fd6ec57572",
       "version-date": "2022-09-05",
       "port-version": 5

